### PR TITLE
Add lowerings for reflection_pad{1, 3}d_backward

### PR DESCRIPTION
**Pull request by isuruf**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #115645

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

### CODE REVIEW

**Comment by isuruf** [comment]
Benchmark results
```
[----------- reflection_padnd_backward ------------]
                             |  Decomposed  |  Eager
32 threads: ----------------------------------------
      (3, 5, 1000000)        |     433      |   795 
      (3, 5, 1000, 1000)     |     434      |   796 
      (3, 5, 100, 100, 100)  |     902      |   975 
```

<details>

```
import torch
from torch.testing import make_tensor
from torch.fx.experimental.proxy_tensor import make_fx
from torch.utils.benchmark import Timer, Compare
from torch._inductor.compile_fx import compile_fx_inner, cudagraphify_impl
from torch._inductor.decomposition import decompositions
from itertools import product
from functools import partial
aten = torch.ops.aten

torch._logging.set_logs(output_code=True)

benchmark_name = "reflection_padnd_backward"
Ss = [[3, 5, 1000000], [3, 5, 1000, 1000], [3, 5, 100, 100, 100]]


def gen_inputs():
    for shape in Ss:
        up_shape = shape[:2] + [s+2 for s in shape[2:]]
        yield [torch.randn(shape, dtype=torch.float32, device="cuda"), torch.randn(up_shape, dtype=torch.float32, device="cuda")]

def benchmark(label, f, x, grad):
    return Timer("f([x, grad])",
                 globals=locals(),
                 label=benchmark_name,
                 description=label,
                 sub_label=f"{tuple(x.shape)}",
                 num_threads=torch.get_num_threads()).blocked_autorange(min_run_time=2)

def compare(args):
    x, grad_output = args
    print(f"{tuple(x.shape)}")
    if x.dim() == 3:
        aten_func = aten.reflection_pad1d_backward
    elif x.dim() == 4:
        aten_func = aten.reflection_pad2d_backward
    elif x.dim() == 5:
        aten_func = aten.reflection_pad3d_backward
    
    def f(args):
        x, grad_output = args
        val = aten_func(grad_output, x, [1]*2*(x.dim() - 2))
        return (val,)

    decomposed = make_fx(f, decomposition_table=decompositions, tracing_mode="fake")(args)
    compiled_decomposed = compile_fx_inner(decomposed, args, cudagraphs=False)
    yield benchmark(f"Decomposed", compiled_decomposed, *args)

    # Just show the first two generated kernels
    #torch._logging.set_logs(output_code=False)

    cuda_f = cudagraphify_impl(f, args, static_input_idxs=tuple(range(len(args))))
    yield benchmark(f"Eager", cuda_f, *args)

results = []
for x in gen_inputs():
    for res in compare(x):
        results.append(res)

compare = Compare(results)
compare.trim_significant_figures()
compare.print()
```

</details>

**Review by lezcano — APPROVED** [review]
Great stuff! Thank you!

File: torch/_inductor/lowering.py [hunk 1]
```diff
@@ -3489,22 +3488,25 @@ def get_x_interp(y):
...
-    left, right, top, bot = padding
+@register_lowering(aten.reflection_pad3d_backward)
+def _reflection_padnd_backward(grad_output, x, padding):
+    dim = len(padding) // 2
 
-    *_, h, w = x.get_size()
-    h = V.graph.sizevars.evaluate_static_shape(h) - 1
-    w = V.graph.sizevars.evaluate_static_shape(w) - 1
+    dhw = x.get_size()[-dim:]
+    dhw = [V.graph.sizevars.evaluate_static_shape(h) - 1 for h in dhw]
```
Comment (peterbell10) [inline torch/_inductor/lowering.py:3498]: I know it's pre-existing but do we need static shapes here? `index_range_condition` even uses `index_expr` so it seems like expressions are supported.

Comment (isuruf) [reply torch/_inductor/lowering.py:3498]: Static shapes are not needed. I checked and it works just fine with symbolic shapes.

**Review by peterbell10 — APPROVED** [review]

**Pull request closed without merging**